src/client.ts

```diff
@@ -22,14 +22,13 @@ interface ClientConfigOptions {
 /**
  * The Client is access point for the embedded UI's in the host application.
  */
-class Client {
-  private _subscriptionManager: SubscriptionManager;
+class Client extends (EventEmitter as { new (): PublicationEventEmitter }) {
```

Comment:
I generally try to opt for composition over extension/inheritance. I think this is a good case where instead of extending `EventEmitter`, we would benefit from creating a new `EventEmitter` as a private property of the class, and delegating calls to it.  That lets us expose less of it's API (see other comments) and be less coupled to a specific implementation.